src/Api/JsonClient.php

```diff
@@ -108,34 +86,14 @@ public function __call($name, array $params)
      */
     protected function postRequest(array $data)
     {
-        $port = $this->useSsl ? 443 : 80;
-        $host = ($this->useSsl ? 'ssl://' : '') . $this->host;
-
-        $socket = @fsockopen($host, $port);
-        if ($socket === false) {
-            $error = error_get_last();
-            throw new \RuntimeException("Failed to connect to {$this->host} on port $port, {$error['message']}");
-        }
+        $socket = $this->getConnection();
```

Comment:
Is there any reason why we can't switch to using curl here as well to make/execute the http requests? Since we're now using it, we could just use it for all requests, it should make this method a lot more simple